File: src/main/java/com/syncstream/syncstream/model/VideoInfo.java
package com.syncstream.syncstream.model;

public class VideoInfo {
    private String url;
    private String title;
    private Long timestamp;

    public VideoInfo(String url, String title, Long timestamp)  {
        this.url = url;
        this.title = title;
        this.timestamp = timestamp;
    }

    public String getUrl() {
        return url;
    }

    public String getTitle() {
        return title;
    }

    public Long getTimestamp() {
        return timestamp;
    }
}

File: syncstream/src/datamodules/video_info.rs
// video_info.rs implements the VideoInfo module

pub struct VideoInfo {
    url: String,
    title: String,
    timestamp: Long,
}

impl VideoInfo {
    fn new(url: String, title: String, timestamp: Long) -> Self {
        VideoInfo {
            url,
            title,
            timestamp,
        }
    }
}
